storage/engine/rocksdb.go

```diff
@@ -166,6 +167,40 @@ func (r *RocksDB) getInternal(key proto.EncodedKey, snapshotHandle *C.DBSnapshot
 	return cStringToGoBytes(result), nil
 }
 
+// GetProto fetches the value at the specified key and unmarshals it.
+func (r *RocksDB) GetProto(key proto.EncodedKey, msg gogoproto.Message) (
+	ok bool, keyBytes, valBytes int64, err error) {
+	return r.getProtoInternal(key, msg, nil)
+}
+
+func (r *RocksDB) getProtoInternal(key proto.EncodedKey, msg gogoproto.Message,
+	snapshotHandle *C.DBSnapshot) (ok bool, keyBytes, valBytes int64, err error) {
+	if len(key) == 0 {
+		err = emptyKeyError()
+		return
+	}
+	var result C.DBString
+	if err = statusToError(C.DBGet(r.rdb, snapshotHandle, goToCSlice(key), &result)); err != nil {
+		return
+	}
+	if result.len <= 0 {
+		return
+	}
+	ok = true
+	if msg != nil {
+		// Make a byte slice that is backed by result.data. This slice
+		// cannot live past the lifetime of this method, but we're only
+		// using it to unmarshal the proto.
+		// data := C.GoBytes(unsafe.Pointer(result.data), result.len)
+		data := (*[0x7fffffff]byte)(unsafe.Pointer(result.data))[:result.len:result.len]
```

Comment:
I don't think we currently have any hard limits on data size, although as a practical matter we'll want to set some limits far below 2GB. I'm not too concerned about hitting the 2GB limit (especially for protobufs; anything that big will probably be a blob instead of a protobuf), I'd just rather not have arbitrary numbers in the code that represent "more than we'll ever need" if we can avoid it.